ocs_ci/deployment/aws.py

```diff
@@ -134,18 +150,190 @@ def destroy_cluster(self, log_level="DEBUG"):
             log_level (str): log level openshift-installer (default: DEBUG)
         """
         super(AWSIPI, self).destroy_cluster(log_level)
+        self.destroy_volumes()
 
-        try:
-            # Retrieve cluster name and AWS region from metadata
-            cluster_name = self.ocp_deployment.metadata.get("clusterName")
-            # Find and delete volumes
-            volume_pattern = f"{cluster_name}*"
-            logger.debug(f"Finding volumes with pattern: {volume_pattern}")
-            volumes = self.aws.get_volumes_by_name_pattern(volume_pattern)
-            logger.debug(f"Found volumes: \n {volumes}")
-            for volume in volumes:
-                self.aws.detach_and_delete_volume(
-                    self.aws.ec2_resource.Volume(volume['id'])
+
+class AWSUPI(AWSBase):
+    """
+    A class to handle AWS UPI specific deployment
+    """
+    def __init__(self):
+        self.name = self.__class__.__name__
+        super(AWSUPI, self).__init__()
+
+    class OCPDeployment(BaseOCPDeployment):
```

Comment:
why is the class inside UPI class?  why can't it be just methods for UPI and use OCP deployment as  composition? I am also failing to understand what the *isa* relation between the classes, can you docstring them clearly please. Thanks

Reply:
This is implementation of the OCPDeployment class for specific platform/deployment type. If it's like that than we can in each type of deployment instantiate the class with the same name which is already done from the base class of the deployment.

It's documented a bit here: https://github.com/red-hat-storage/ocs-ci/blob/master/ocs_ci/deployment/deployment.py#L46

Base class is here: https://github.com/red-hat-storage/ocs-ci/blob/master/ocs_ci/deployment/ocp.py#L22

Reply:
OCP Deployment is different from OCS Deployment so they dont really have to inherit and should be kept as different classes, Also the use of inner classes seems like overkill now. we could just use simple classes and composition for better readability here and maintain relationship based on strict inheritance eg:  https://www.w3resource.com/java-tutorial/inheritance-composition-relationship.php

Reply:
@vasukulkarni I agree, and think we can simplify these classes a lot by not having them nested. It's not something we are going to address in this PR, but i think it's something we should absolutely create an issue for and discuss.

Reply:
@clacroix12 sure, I will try to add it in issues today or monday.